addons/web/controllers/view.py

```diff
@@ -25,3 +34,43 @@ def edit_custom(self, custom_id, arch):
             ))
         custom_view.write({'arch': arch})
         return {'result': True}
+
+    @route('/web/view/save_multi_with_operations', type='jsonrpc', auth="user")
+    def save_multi_with_operations(self, model, ids, changes, context=None):
+        if not model or not ids or not isinstance(changes, dict):
+            return {'error': 'Invalid parameters'}
+
+        try:
+            env = request.env[model].with_context(context or {})
```

Comment:
this is overriding all keys from initiial context
also, you are not getting an env, but an empy recordset
that line should become:
```py
Model = request.env[model].with_context(**(context or {}))

```